### PR TITLE
Enable IPV6 config on Sandbox creation on live-restore case

### PR DESCRIPTION
**Pull request by selansen**
This is to address issue https://github.com/moby/moby/issues/35757.

In sandbox creation we disable IPV6 config. But this causes problem in live-restore case
where all IPV6 configs are wiped out on running container. Hence extra check has been added
take care of this issue.

Signed-off-by: selansen <elango.siva@docker.com>

### CODE REVIEW

**Comment by mavenugo** [comment]
LGTM

Can you pls write a test case for this ?

**Comment by fcrisciani** [comment]
LGTM too, waiting for the test

File: osl/namespace_linux.go [hunk 1]
```diff
@@ -220,15 +220,18 @@ func NewSandbox(key string, osCreate, isRestore bool) (Sandbox, error) {
...
-
-	// As starting point, disable IPv6 on all interfaces
-	if !n.isDefault {
-		err = setIPv6(n.path, "all", false)
-		if err != nil {
-			logrus.Warnf("Failed to disable IPv6 on all interfaces on network namespace %q: %v", n.path, err)
+	// In live-restore mode, IPV6 entries are getting cleaned up due to below code
+	// We should retain IPV6 configrations in live-restore mode when Docker Daemon
+	// comesback. It should work as it is on other cases
+	if !isRestore {
```
Comment (aboch) [inline osl/namespace_linux.go]: Can you please avoid the double indentation of the logic below by instead

```
if !isRestore && !n.isDefault {
```

Thanks

Comment (thaJeztah) [reply osl/namespace_linux.go]: ping @selansen ^^

**Comment by selansen** [comment]
I made the changes but trying to come up with unit test for this case and
its kind of  tricky to recreate this scenario with unit test case.  will
try to commit with basic unit test for live-restore true case while
creating sandbox and commit it.

**Comment by selansen** [comment]
Incorporated all comments and added new unit test case . Pls merge the commit into upstream.

**Comment by fcrisciani** [comment]
LGTM

**Pull request merged**